source/common/http/conn_manager_impl.h

```diff
@@ -421,32 +462,72 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
       }
     }
 
+    /**
+     * Decodes the provided headers starting at the first filter in the chain.
+     * @param headers the headers to decode.
+     * @param end_stream whether the request is header only.
+     */
     void decodeHeaders(RequestHeaderMap& headers, bool end_stream) {
       decodeHeaders(nullptr, headers, end_stream);
     }
+
+    /**
+     * Decodes the provided data starting at the first filter in the chain.
+     * @param data the data to decode.
+     * @param end_stream whether this data is the end of the request.
+     */
     void decodeData(Buffer::Instance& data, bool end_stream) {
       decodeData(nullptr, data, end_stream, FilterIterationStartState::CanStartFromCurrent);
     }
+
+    /**
+     * Decodes the provided trailers starting at the first filter in the chain.
+     * @param trailers the trailers to decode.
+     */
     void decodeTrailers(RequestTrailerMap& trailers) { decodeTrailers(nullptr, trailers); }
+
+    /**
+     * Decodes the provided metadata starting at the first filter in the chain.
+     * @param metadata_map the metadata to decode.
+     */
     void decodeMetadata(MetadataMap& metadata_map) { decodeMetadata(nullptr, metadata_map); }
 
+    // TODO(snowp): Make private as filter chain construction is moved into FM.
     void addStreamDecoderFilterWorker(StreamDecoderFilterSharedPtr filter, bool dual_filter);
     void addStreamEncoderFilterWorker(StreamEncoderFilterSharedPtr filter, bool dual_filter);
 
     void disarmRequestTimeout();
+
+    /**
+     * If end_stream is true, marks decoding as complete. This is a noop if end_stream is false,
+     * which helps DRY up the if checks.
```

Comment:
not familiar with DRY up in this context?

Reply:
I'll just remove this part, its just an explanation why we use this pattern instead of `if (end_stream) { endEncode(); }` all over the place